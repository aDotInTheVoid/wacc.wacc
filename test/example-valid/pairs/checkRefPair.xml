<unit>
  <main>
    <addr_local>0</addr_local>
    <num>10</num>
    <char>'a'</char>
    <newpair/>
    <assign_do/>
    <addr_local>1</addr_local>
    <local>0</local>
    <assign_do/>
    <local>0</local>
    <print><kind>ptr</kind><newline>true</newline></print>
    <local>1</local>
    <print><kind>ptr</kind><newline>true</newline></print>
    <local>0</local>
    <local>1</local>
    <op>eq</op>
    <print><kind>bool</kind><newline>true</newline></print>
    <addr_local>2</addr_local>
    <local>0</local>
    <fst/>
    <assign_do/>
    <addr_local>3</addr_local>
    <local>1</local>
    <fst/>
    <assign_do/>
    <local>2</local>
    <print><kind>i32</kind><newline>true</newline></print>
    <local>3</local>
    <print><kind>i32</kind><newline>true</newline></print>
    <local>2</local>
    <local>3</local>
    <op>eq</op>
    <print><kind>bool</kind><newline>true</newline></print>
    <addr_local>4</addr_local>
    <local>0</local>
    <snd/>
    <assign_do/>
    <addr_local>5</addr_local>
    <local>1</local>
    <snd/>
    <assign_do/>
    <local>4</local>
    <print><kind>char</kind><newline>true</newline></print>
    <local>5</local>
    <print><kind>char</kind><newline>true</newline></print>
    <local>4</local>
    <local>5</local>
    <op>eq</op>
    <print><kind>bool</kind><newline>true</newline></print>
  </main>
</unit>