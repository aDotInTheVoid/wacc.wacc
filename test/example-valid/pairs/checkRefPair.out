0x{{PTR}}
0x{{PTR}}
true
10
10
true
a
a
true
